source/common/secret/secret_manager_impl.cc

```diff
@@ -155,17 +156,23 @@ void redactSecret(envoy::extensions::transport_sockets::tls::v3alpha::Secret* se
 }
 
 ProtobufTypes::MessagePtr SecretManagerImpl::dumpSecretConfigs() {
+  // TODO(htuch): unlike other config providers, we're recreating the original
+  // Secrets below. This makes it hard to support API_RECOVER_ORIGINAL()-style
+  // recovery of the original config message. As a result, for now we're
+  // providing v3 config dumps. For Secrets, the main deprecation of interest
+  // are the use of v2 Struct config() and verify_subject_alt_name.
```

Comment:
Why are we doing this? Should we just fix this to store the original message? We can do this in a follow up?

Reply:
Yep, tracked in https://github.com/envoyproxy/envoy/issues/9670.